dna/compare.go

```diff
@@ -5,20 +5,43 @@ import (
 	"log"
 )
 
+// compareBases returns an integer related to the lexographical order of nucleotides.
+// i.e. A < C < a < c < Dot < Gap
 func compareBases(alpha Base, beta Base, ignoreCase bool) int {
+	var a, b rune
 	if ignoreCase {
-		alpha = ToUpper(alpha)
-		beta = ToUpper(beta)
+		a = rune(ToUpper(alpha))
+		b = rune(ToUpper(beta))
+	} else {
+		a = rune(alpha)
+		b = rune(beta)
 	}
-	if alpha < beta {
-		return -1
-	} else if alpha > beta {
-		return 1
+
+	if alpha == Gap || alpha == Dot || beta == Gap || beta == Dot {
```

Comment:
I am not sure why we switch 1 and -1 when either alpha or beta are a dash.  Is it so that A < -?  Why not follow ascii convention?